src/transformers/modeling_utils.py

```diff
@@ -730,7 +730,9 @@ def generate(
             assert input_ids.dim() == 2, "Input prompt should be of shape (batch_size, sequence length)."
 
         # current position and vocab size
-        cur_len = input_ids.shape[1]
+        # cur_len = input_ids.shape[1] TODO FOR SEQ-2-SEQ: This doesn't work for seq-2-seq models as the input_ids will be encoded and not prepended to the output. Proposal:
+        cur_len = 0 if self._is_seq_to_seq() else input_ids.shape[0]
```

Comment:
_is_seq_to_seq should probably not start with `_`.
There is also `is_encoder_decoder` in the ModelTester, and I feel like it should be one param :)